src/project.rs

```diff
@@ -266,6 +266,36 @@ impl FuzzProject {
         Ok(cmd)
     }
 
+    fn target_dir(&self, build: &options::BuildOptions) -> Result<PathBuf> {
+        if build.coverage {
+            Ok(env::current_dir()?
+                .join("target")
+                .join(default_target())
+                .join("coverage"))
+        } else {
+            Ok(build
+                .target_dir
+                .as_ref()
+                .map(PathBuf::from)
+                .unwrap_or_else(|| self.project_dir.join("target")))
```

Comment:
If the user explicitly passes in a `--target-dir` I think that should take priority over the default coverage target directory. So I think this should look something like:

```rust
if let Some(target_dir) = build.target_dir {
    // use user-provided target dir...
} else if build.coverage {
    // use coverage target dir...
} else {
    // use default target dir...
}
```